src/redisearch_rs/rlookup/src/lookup.rs

```diff
@@ -239,6 +266,392 @@ impl<'a> RLookupKey<'a> {
         // Safety: 3 -> Caller has to uphold the pin contract
         unsafe { Pin::new_unchecked(b) }
     }
+
+    /// Returns `true` if this node is currently linked to a [`List`].
+    #[cfg(test)]
+    fn has_next(&self) -> bool {
+        self.next().is_some()
+    }
+
+    /// Return the next pointer in the linked list
+    #[inline]
+    fn next(&self) -> Option<NonNull<RLookupKey<'a>>> {
+        // Safety: RLookupKeys are created through `KeyList::push` and owned by the `List`. We
+        // can therefore assume this pointer is safe to dereference at this point.
+        unsafe { *self.next.get() }
+    }
+
+    /// Update the pointer to the next node
+    #[inline]
+    fn set_next(
+        &mut self,
+        next: Option<NonNull<RLookupKey<'a>>>,
+    ) -> Option<NonNull<RLookupKey<'a>>> {
+        mem::replace(self.next.get_mut(), next)
+    }
+
+    fn assert_valid(&self, tail: &Self, ctx: &str) {
```

Comment:
Can be conditioned on `#[cfg(debug_assertions)]` ?

Reply:
or test

Reply:
I made the function itself explicitly not `#[cfg(debug_assertions)]` because I figured we might want the option to assert consistency at very specific points even in release builds. 
But at the moment at least all *callsites* of this function are guarded behind `#[cfg(debug_assertions)]` so this will only ever be called in `debug` mode.

Edit: I'm happy to add the `cfg` to this one too, its easy to remove later if we end up needing.